scipy/cluster/hierarchy.py

```diff
@@ -619,55 +618,53 @@ def linkage(y, method='single', metric='euclidean'):
     Z : ndarray
         The hierarchical clustering encoded as a linkage matrix.
 
+    Notes
+    -----
+    1. For method 'single' an optimized algorithm called SLINK is implemented,
+       which has :math:`O(n^2)` time complexity and :math:`O(n)` memory
```

Comment:
The space complexity is in fact O(n^2) in our current implementation, as we pass a `dist` array which has n(n-1)/2 elements, into the linkage function. An O(n) space implementation of SLINK requires computing distances on the fly, which seems not to fit well into our API design.